pkg/cmd/roachtest/test_impl.go

```diff
@@ -548,6 +549,68 @@ func failuresMatchingError(failures []failure, refError any) bool {
 	return false
 }
 
+// transientErrorOwnershipFallback string matches for `TRANSIENT_ERROR` in the error
+// message and returns a new ErrorWithOwnership if it does.
+//
+// This is needed as the `require` package does not preserve the error object needed
+// for us to properly check if it is a transient error using `failuresMatchingError`.
+// Note we only check if we also see requireNoErrorPrefix, indicating it's an error reported
+// by the require package. If we see somewhere else that does not preserve the error object,
+// we want to investigate whether it can be fixed before resorting to this fallback.
+// See: #131094
+func transientErrorOwnershipFallback(failures []failure) *registry.ErrorWithOwnership {
+	const (
+		transientErrorRegex  = "TRANSIENT_ERROR\\((.+)\\)"
+		requireNoErrorPrefix = "Received unexpected error:"
+	)
+
+	regex, err := regexp.Compile(transientErrorRegex)
```

Comment:
Nit: the cost is O(len(transientErrorRegex)) * numTests; could lift into a global to reduce to O(len(transientErrorRegex)).